src/main/java/com/jwplayer/jwplatform/client/MediaClient.java

```diff
@@ -0,0 +1,59 @@
+package com.jwplayer.jwplatform.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.json.JSONObject;
+
+import com.jwplayer.jwplatform.exception.JWPlatformException;
+import com.jwplayer.jwplatform.rest.HttpCalls;
+
+public class MediaClient{
+
+	private String path;
+	private String secret;
+	
+	public MediaClient(String secret) {
+		this.secret = secret;
+		this.path = "https://api.jwplayer.com/v2/sites/%s/media/";
+	}
+	
+	private String format(String siteId, Map<String,String> headers) {
```

Comment:
in general, try to stick with pure functions. This mutation of headers is buried and could lead to debugging issues.

Just make `headers` a static map within the class since its the same for all endpoints. Since you are not exposing headers to the user, I assume there is no instance where the user would ever need to add a header? I ask b/c my team actually uses this lib in another app and pass `x-jw-ratelimit-bypass` as an additional header.